ietf-optical-impairment-topology.tree

```diff
@@ -49,39 +56,108 @@ module: ietf-optical-impairment-topology
-    +--ro OTSiG-element* [OTSiG-identifier]
-    |  +--ro OTSiG-identifier    int16
-    |  +--ro OTSiG-container
-    |     +--ro OTSi* [OTSi-carrier-id]
-    |        +--ro OTSi-carrier-id           int16
-    |        +--ro OTSi-carrier-frequency?   decimal64
-    |        +--ro OTSi-signal-width?        decimal64
-    |        +--ro channel-delta-power?      decimal64
-    +--ro transponders-list* [transponder-id]
-       +--ro transponder-id                      uint32
-       +--ro (mode)?
-       |  +--:(G.692.2)
-       |  |  +--ro standard-mode?                standard-mode
-       |  +--:(organizational-mode)
-       |  |  +--ro operational-mode?             operational-mode
-       |  |  +--ro organization-identifier?      vendor-identifier
-       |  +--:(explicit-mode)
-       |     +--ro available-modulation-types*   identityref
-       |     +--ro configured-modulation-type?   identityref
-       |     +--ro available-baud-rates*         uint32
-       |     +--ro configured-baud-rate?         uint32
-       |     +--ro available-FEC-types*          identityref
-       |     +--ro configured-FEC-type?          identityref
-       |     +--ro FEC-code-rate?                decimal64
-       |     +--ro FEC-threshold?                decimal64
-       +--ro power?                              int32
-       +--ro power-min?                          int32
-       +--ro power-max?                          int32
-  augment /nw:networks/nw:network/nw:node/tet:te/tet:tunnel-termination-point:
-    +--ro transponder-list* [carrier-id]
+  augment /nw:networks/nw:network/nw:node/tet:te
+            /tet:tunnel-termination-point:
+    +--ro otsi-group* [otsi-group-id]
+    |  +--ro otsi-group-id    int16
+    |  +--ro otsi* [otsi-carrier-id]
+    |     +--ro otsi-carrier-id           int16
+    |     +--ro transponder-ref?          leafref
+    |     +--ro transceiver-ref?          leafref
+    |     +--ro configured-mode?          leafref
+    |     +--ro OTSi-carrier-frequency?   frequency-thz
+    |     +--ro tx-channel-power?         dbm-t
+    |     +--ro rx-channel-power?         dbm-t
+    |     +--ro rx-total-power?           dbm-t
+    +--ro transponder* [transponder-id]
+       +--ro transponder-id    uint32
+       +--ro transceiver* [transceiver-id]
+          +--ro transceiver-id     uint32
+          +--ro supported-modes
+             +--ro supported-mode* [mode-id]
+                +--ro mode-id                      string
+                +--ro (mode)
+                   +--:(G.698.2)
+                   |  +--ro standard-mode?         standard-mode
+                   +--:(organizational-mode)
+                   |  +--ro organizational-mode
+                   |     +--ro operational-mode?
+                   |     |       operational-mode
+                   |     +--ro organization-identifier?
+                   |     |       organization-identifier
+                   |     +--ro min-central-frequency?
```

Comment:
Organizational Mode has been defined as:
Organizational Mode: An organizational mode represents a non-standard optical interface specification towards the realization of transversely compatible DWDM systems. Two transceivers supporting the same organizational mode and a line system matching the constraints, defined by the organization which owns the mode, for that organizational will interoperate. These organizations can be MSA-Groups, Operators, System vendors, component vendors etc.
Adding explicit arameters here would defeat the purpose. Explicit parameters are in the explicit mode and the organizational mode can be referenced from there.

Reply:
The organizational mode does not cover all transceiver properties by intention - see: https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/issues/29#issuecomment-718774926
Carrier frequency and power attributes have been excluded because they are totally independent from the transceiver presets defining specific operational modes. See also https://github.com/ietf-ccamp-wg/draft-ietf-ccamp-optical-impairment-topology-yang/issues/29#issuecomment-610288875
Your statement: "Adding explicit parameters here would defeat the purpose. " is incorrect.